src/library/autodj/autodjfeature.cpp

```diff
@@ -308,8 +310,22 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
         retriveAttempts += 1;
     }
 
-    if (retriveAttempts == kMaxRetiveAttempts)
-        qDebug () << "Could Not Load Random Tracks";
+    if (retriveAttempts == kMaxRetiveAttempts || iTrackId == -1) {
+        qDebug () << "Could Not Load Random Tracks From Crate."
+    	        " Attempting to load from library";
+    	iTrackId = m_autoDjCratesDao.getRandomTrackIdFromLibrary();
```

Comment:
The intention here was to prevent the autoDj from stopping abruptly even in the cases where crates exist but tracks couldn't be loaded from them . What you are saying , as I understand is to replace the condition with something like  : get tracks from library if no crates are there ?

This way seems to cover up for more cases ? What do you think ?